dbt_sugar/core/connectors/base.py

```diff
@@ -0,0 +1,55 @@
+"""
+Module base connector.
+
+Only use this class implemented by a child connector.
+"""
+import logging
+from typing import Any, List, Optional, Tuple
+
+
+class BaseConnector:
```

Comment:
Might be a good idea to make this be an ABC right? So that it actually throws proper warnings when the concrete classes don't implement the methods we want?

```python
from abc import ABC, abstractmethod

class BaseConnector(ABC):
    @abstractmethod
    def method_to_implement()
		...
```

What do you think?


Also, I would be of the opinion to implement a separate class for the Connection and one for the implementation so that we can separate them more.

The "connector" would hold the methods to talk to the db and would require a `Connection` object to be passed. I do it a bit like this in sheetwork: https://github.com/bastienboutonnet/sheetwork/tree/main/sheetwork/core/adapters

It would allow us to test things more independently especially if we use the `pytest` fixtures which actually expose a connection object which would be easily mockable.